src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -18,15 +19,19 @@ public class TagHelperContext
         /// Instantiates a new <see cref="TagHelperContext"/>.
         /// </summary>
         /// <param name="allAttributes">Every attribute associated with the current HTML element.</param>
+        /// <param name="parentItems">Items that have been created by parent <see cref="ITagHelper"/>s.</param>
         /// <param name="uniqueId">The unique identifier for the source element this <see cref="TagHelperContext" /> 
         /// applies to.</param>
         /// <param name="getChildContentAsync">A delegate used to execute and retrieve the rendered child content 
         /// asynchronously.</param>
-        public TagHelperContext([NotNull] IDictionary<string, object> allAttributes,
-                                [NotNull] string uniqueId,
-                                [NotNull] Func<Task<string>> getChildContentAsync)
+        public TagHelperContext(
+            [NotNull] IDictionary<string, object> allAttributes,
+            [NotNull] IDictionary<string, object> parentItems,
```

Comment:
That feels pretty heavy handed and seems to defeat the purpose of the `TagHelperExecutionContext` for the `Items` dictionary. I don't see the value in having the `TagHelperContext` own the `CopyOnWriteDictionary` wrapped items. In the end `TagHelperContext` and `TagHelperOutput` are just utility classes sent into `TagHelper` process methods where `TagHelperExecutionContext` is the real state holder behind the scenes.

One thing I want to change one of these days is to no longer expose `TagHelperOutput` to the `RazorPage`, that way users can easily replace that part of `TagHelper` execution, for now that's pretty much the only thing that comes back out into the `RazorPage`.